### ISSUE TITLE
Populate should allow empty relationship field

### ISSUE THREAD

**Issue by Mattchewone** [feathersjs/feathers-hooks-common#138]
### Steps to reproduce

The issue I am having is I have a model called Tags, which has a field called parent which references a tag item. But there will be tags which either have no parent field or the parent field is empty. Currently this throws an error.

```
{
  "name": "BadRequest",
  "message": "ParentField parent for category depth 1 is undefined. (populate)",
  "code": 400,
  "className": "bad-request",
  "errors": {}
}
```

### Expected behavior
Should populate the items which have the related field.

### Actual behavior
Throws an error as above.

### System configuration

**Module versions**:
`feathers-hooks-common@2.0.3`

### A fix
For now after chatting with `eddystop` on slack, he suggested that I populate the field with a null value before the populate hook is run, in the after hook event. 

Could this be built into the populate hook to do itself?

**Comment by Mattchewone** [comment]
I am not sure what impact this may have on other databases but within the `populate.js` if this is removed then this allows the items to be populated. 
```
      if (parentVal === undefined) {
        throw new errors.BadRequest(
          `ParentField ${childSchema.parentField} for ${nameAs} depth ${depth} is undefined. (populate)`
        );
      }
```

With Mongoose the query runs fine and the ones which do not contain the field that is being referenced are still returned just without the relationship, which is exactly what I am looking for, and the behaviour I would expect.

**Comment by ekryski** [comment]
Agreed, I'm also running into this where the field we want to populate on may or may not exist. @Mattchewone the behavior you described above is exactly how I would expect this to work and we're using MongoDB as well.

@eddyystop any thoughts on whether removing that check will cause issues with other DBs?

**Comment by eddyystop** [comment]
This change already has a `scheduled` tag.

The only issue I see is developers not finding the array of items they thought should be joined and taking to Slack or Stackoverflow.

**Comment by dottodot** [comment]
Making the value null if missing works and prevents errors but does have some strange results as it returns an empty array even with `asArray = false`

**Comment by ellipticaldoor** [comment]
I have a solution but it's not very elegant

``` javascript
exports.after = {
	all: [
		hooks.remove('_id'),

		// This is an ugly fix to allow to populate an field that maybe doesn't exists
		hook => {
			const results = hook.result.data

			if (results) {
				for (const result of results)
					if (result.to === undefined)
						result.to = null
			}
			else
				if (hook.result.to === undefined)
					hook.result.to = null
		},

		hooks.populate({
			schema: {
				include: [
					{
						service: 'users',
						nameAs: 'from',
						parentField: 'from',
						childField: '_id',
						query: {
							$select: ['_id', 'name'],
						},
					},
					{
						service: 'users',
						nameAs: 'to',
						parentField: 'to',
						childField: '_id',
						query: {
							$select: ['_id', 'name'],
						},
					},
				],
			},
		}),
	],
}
```

**Comment by eddyystop** [comment]
A proper fix is included in https://github.com/feathersjs/feathers-hooks-common/pull/170 . It will be published as v3.2.1.
